torch/csrc/autograd/function.h

```diff
@@ -90,9 +139,32 @@ struct Function : std::enable_shared_from_this<Function> {
   };
 
   // Computes is_executable and next_functions from a list of input variables
-  static FunctionFlags flags(const variable_list& inputs);
-  static FunctionFlags flags(const std::initializer_list<Variable>& inputs);
-  static FunctionFlags flags(at::TensorList inputs);
+  // (but which are tensors, because this is what happens in the autogenerated
+  // code)
+  template<typename... Args> inline static FunctionFlags tensor_flags(Args... args) {
+    FunctionFlags f;
+    f.is_executable = false;
+    f.next_functions.resize(countTensors(args...));
+    if (!GradMode::is_enabled()) {
+      // TODO: avoid allocating next_functions entirely if grad_mode is disabled
+      return f;
+    }
+    set_tensor_function_flags(f, 0, args...);
+    return f;
+  }
+
+  // Computes is_executable and next_functions from a list of input variables
+  template<typename... Args> inline static FunctionFlags flags(Args... args) {
+    FunctionFlags f;
+    f.is_executable = false;
+    f.next_functions.resize(countVariables(args...));
+    if (!GradMode::is_enabled()) {
+      // TODO: avoid allocating next_functions entirely if grad_mode is disabled
```

Comment:
Why can't you just move the line that resizes it below this `if`?

Reply:
I could have said the same thing for the original code. Since such a thing was not done, I assumed that there was some reason why it wasn't as simple as that.